cmd/eval.go

```diff
@@ -220,31 +236,38 @@ func determineInputSpec(arguments args) (*appstudioshared.ApplicationSnapshotSpe
 }
 
 func validateImage(ctx context.Context, imageRef, policyConfiguration, publicKey, rekorURL string) (*policy.Output, error) {
-	out := &policy.Output{}
+	imageValidator, err := image.NewImageValidator(ctx, imageRef, publicKey, rekorURL)
+	if err != nil {
+		return nil, err
+	}
 
-	i, err := image.NewImageValidator(ctx, imageRef, publicKey, rekorURL)
+	policyEvaluator, err := policy.NewPolicyEvaluator(policyConfiguration)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := i.ValidateImageSignature(ctx); err != nil {
+	return validateImageWith(ctx, imageRef, policyConfiguration, publicKey, rekorURL, imageValidator, policyEvaluator)
+}
+
+func validateImageWith(ctx context.Context, imageRef, policyConfiguration, publicKey, rekorURL string, imageValidator image.ImageValidator, policyEvaluator policy.PolicyEvaluator) (*policy.Output, error) {
+	out := &policy.Output{}
+	if err := imageValidator.ValidateImageSignature(ctx); err != nil {
 		out.SetImageSignatureCheck(false, err.Error())
-		return nil, err
+	} else {
+		out.SetImageSignatureCheck(true, "success")
 	}
-	out.SetImageSignatureCheck(true, "success")
 
-	if err := i.ValidateAttestationSignature(ctx); err != nil {
+	if err := imageValidator.ValidateAttestationSignature(ctx); err != nil {
 		out.SetAttestationSignatureCheck(false, err.Error())
-		return nil, err
+	} else {
+		out.SetAttestationSignatureCheck(true, "success")
 	}
-	out.SetAttestationSignatureCheck(true, "success")
 
-	p, err := policy.NewPolicyEvaluator(policyConfiguration)
-	if err != nil {
-		return nil, err
+	if !out.ImageSignatureCheck.Passed || !out.AttestationSignatureCheck.Passed {
```

Comment:
Technically, we don't need the `ImageSignatureCheck` to have passed to perform the policy evaluation. `AttestationSignatureCheck` is the only requirement.

Reply:
I think we could check if there are any attestations present